demisto_sdk/commands/format/tests/test_formatting_yml_test.py

```diff
@@ -598,6 +598,44 @@ def test_update_docker_format(tmpdir, mocker):
     assert data['script']['dockerimage'].endswith(f':{test_tag}')
 
 
+def test_update_docker_format_with_invalid_dockerimage(requests_mock, mocker, tmp_path):
+    """Test that script and integration formatter update docker image tag
+    """
+
+    auth_token = 'ABCDE12345'
+    mocker.patch.object(DockerImageValidator, 'docker_auth', return_value=auth_token)
+
+    requests_mock.get('https://hub.docker.com/v2/repositories/error/tags', json={"detail": "Object not found"},
+                      status_code=404)
+    requests_mock.get('https://registry-1.docker.io/v2/error/tags/list', json={'error': 'not found'},
+                      status_code=401)
+    requests_mock.get('https://hub.docker.com/v2/repositories/demisto/error/tags', json={"count": 0,
+                                                                                         "next": 'null',
+                                                                                         "previous": 'null',
+                                                                                         "results": []},
+                      status_code=200)
+    integration_yml_file_1 = tmp_path / 'Integration1.yml'
+    integration_obj = {'script': {'dockerimage': 'error:1.0.0.1'}}
+    ryaml.dump(integration_obj, integration_yml_file_1.open('w'))
+
+    format_obj = ScriptYMLFormat(str(integration_yml_file_1), update_docker=True)
+    format_obj.update_docker_image()
+    with open(str(integration_yml_file_1)) as f:
+        data = yaml.safe_load(f)
+    assert data.get('script', {}).get('dockerimage') == 'error:1.0.0.1'
+
+    # test example script file with invalid docker image : demisto/error
+    integration_yml_file_2 = tmp_path / 'Integration2.yml'
```

Comment:
dont you need different requests mocks for the second case?

Reply:
Yes, Two different cases, fails at different requests.